### ISSUE TITLE
Support for striketrough text on write_html function

### ISSUE THREAD

**Issue by viniciustozzi** [py-pdf/fpdf2#1322]
**Please explain your intent**
It would be greate to have support for striketrough text on `write_html` function. Currently it already works for bold, italic and underline. 

**Describe the solution you'd like**
When using `write_html()` function it is already possible to use `tag_styles` argument to map the tag `<s>` to a `TextStyle` or `FontFace`, but unfortunately there is no way to set the emphasis to striketrough. Maybe adding a new Text Emphasis for striketrough  ("S") would be enough. However I am not sure how to actually render this on the PDF itself.

**Additional context**
With the following snippet it's possible to see that `<b>` tag works just fine but when I try with `<s>` tag it does not work:

```python
from fpdf import FPDF

pdf = FPDF()
pdf.write_html(
    text="<b>Bold Text works </b> <s> but strikethrough does not </s>.",
)
pdf.output("issue_1322.pdf")
```

**Comment by Lucas-C** [comment]
Hi @viniciustozzi

Sure, this an excellent suggestion for a feature!

Pull Requests are welcome to implement this 🙂

First, we could either implement this in [`HTML2FPDF`](https://github.com/py-pdf/fpdf2/blob/master/fpdf/html.py#L304), or else in the base `FPDF` class, similarly to how "underlined text" is implemented.

If we chose the second option, "strikethrough" should be part of the [`GraphicsState`](https://github.com/py-pdf/fpdf2/blob/master/fpdf/graphics_state.py).
But another question arise: should it be included in [TextEmphasis](https://github.com/py-pdf/fpdf2/blob/master/fpdf/enums.py#L222), that currently combines bold, italics & underline text styling?

IMHO, yes, we should implement strikethrough in exatcly the same way that underline is currently implemented.
But it would be very interesting to have other people view on this 🙂
Ping other maintainers @andersonhc & @gmischler

**Comment by andersonhc** [comment]
I agree it would be a nice addition and we should implement strikethrough like underline.

We load underline thickness and position from the font's "post" table - with hardcoded values for the core fonts.
I believe the equivalent for strikethrough would be yStrikeoutPosition and yStrikeoutSize on the OS/2 table.

**Comment by Lucas-C** [comment]
I opened PR https://github.com/py-pdf/fpdf2/pull/1340 as a starting point.

It does not seem that yStrikeoutPosition / yStrikeoutSize are always present in OS/2 tables.
I will investigate further 🙂

**Comment by Lucas-C** [comment]
I figured it out, I was using the wrong table in the font 😅

The PR is ready for review

**Comment by Lucas-C** [comment]
This has released now in https://github.com/py-pdf/fpdf2/releases/tag/2.8.3
